packages/client/src/schema/repository.ts

```diff
@@ -1655,15 +1667,29 @@ export class RestRepository<Record extends XataRecord>
   }
 }
 
-const transformObjectLinks = (object: any) => {
+function cleanSaveRecord<T>(object: any, tableColumns: Schemas.Column[]): T {
```

Comment:
I'm not sure about this. The fact that we ignore fields that are not in the database could be a source of confusion.

I think we can maybe expose this function with a different name and allow users to use it explicitely, but not do it internally ourselves.

WDYT?

Reply:
I would like to avoid adding even more methods, but an option `ignoreExtraProps` might be helpful

Reply:
`ignoreExtraProps` works for me.